code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -47,13 +47,15 @@
 	return on_consumption(M, user)
 
 ///Runs the consumption code, can be overriden for special effects
-/obj/item/reagent_containers/pill/proc/on_consumption(mob/M, mob/user)
-	if(icon_state == "pill4" && prob(5)) //you take the red pill - you stay in Wonderland, and I show you how deep the rabbit hole goes
-		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), M, span_notice("[pick(strings(REDPILL_FILE, "redpill_questions"))]")), 50)
+/obj/item/reagent_containers/pill/proc/on_consumption(mob/consumer, mob/giver)
+	if(icon_state == "pill4") //you take the red pill - you stay in Wonderland, and I show you how deep the rabbit hole goes
+		SEND_SIGNAL(consumer, COMSIG_BITRUNNER_RED_PILL_SEVER) //if your bitrunning avatar somehow manages to acquire and consume a red pill, they will be ejected from the Matrix
```

Comment:
I'd prefer if this were a generic `COMSIG_MOB_PILL_CONSUMED` than a signal specific to bitrunners. You can move the icon state check to a new signal proc for the avatar connection component.

Reply:
is there any feasible use case for such a generic "pill consumed" signal, other than "hehe funny matrix reference"?

the icon state check is still needed for the other red pill effect, which I don't want to touch with this PR or to tie to bitrunning.

Reply:
@Ghommie

Reply:
> is there any feasible use case for such a generic "pill consumed" signal, other than "hehe funny matrix reference"?

Right now, it isn't be the case, but in the future, it might has well have more uses. It doesn't hurt to have a generic signal.

> the icon state check is still needed for the other red pill effect, which I don't want to touch with this PR or to tie to bitrunning.

Oh yeah, keep that!

Reply:
> > is there any feasible use case for such a generic "pill consumed" signal, other than "hehe funny matrix reference"?
> 
> Right now, it isn't be the case, but in the future, it might has well have more uses. It doesn't hurt to have a generic signal.

Shouldn't we wait to add such a signal until the PR that adds those other uses?

Reply:
this isn't how signals are supposed to be used, they aren't proc calls, they are events

Reply:
isn't this how the other bitrunning ejection triggers are done?

Reply:
>is there any feasible use case for such a generic "pill consumed" signal

You want to make an item that scans pills as you eat them, maby you want to make a spell that makes consumed pills end up in some one else. Just check logic. if you make a signal its best to make it generic. Thats the point after all